valet/pathproc.go

```diff
@@ -42,35 +43,41 @@ type ProcessParams struct {
 	MaxProc       int           // The maximum number of threads to run
 }
 
-func ProcessFiles(cancelCtx context.Context, params ProcessParams) {
+func ProcessFiles(cancelCtx context.Context, params ProcessParams) error {
 
 	wpaths, werrs := WatchFiles(cancelCtx, params.Root, params.MatchFunc,
```

Comment:
WatchFiles and FindFilesInterval use cancelCtx to stop looping, but don't close their output channels. Doesn't that mean that DoProcessFiles just keeps waiting on cancellation?

Reply:
Actually, they do close